docs(platformAutomerge): recommend setting required status checks (#16965)
renovatebot/renovate: docs/usage/configuration-options.md

## platformAutomerge

<!-- prettier-ignore -->
!!! warning

>     Before you enable `platformAutomerge` you must tell your Git hosting platform what test(s) must pass before the PR is allowed to be auto-merged.

Reviewer: ```suggestion
    Before you enable `platformAutomerge` you should enable your Git hosting platform's capabilities to enforce test passing before PR merge.
```

>     If you don't do this, Renovate auto-merges PRs even if all test fail!

Reviewer 2: ```suggestion
    If you don't do this, The platform auto-merges PRs even if all test fail!
```
the platform will do the auto-merge

>     If you don't do this, Renovate auto-merges PRs even if all test fail!

Reviewer: ```suggestion
    If you don't do this, the platform might merge Renovate PRs even if the repository's tests haven't started, are in still in progress, or possibly even when they have failed. 
```

    On GitHub this is called "Require status checks before merging", which you can find in the "Branch protection rules" section of the settings for your repository.
    [GitHub docs, about protected branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches)
    [GitHub docs, require status checks before merging](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging)

>     If you're using another platform, search their documentation for a similar feature.

Author: I don't know how to format this text so:

- Prettier ignores the formatting
- Material for MkDocs renders the admonition correctly on the published page

Here's what the end-result should look like:

> Before you enable `platformAutomerge` you must tell your Git hosting platform what test(s) must pass before the PR is allowed to be auto-merged.
> If you don't do this, Renovate auto-merges PRs even if all test fail!
> 
> On GitHub this is called "Require status checks before merging", which you can find in the "Branch protection rules" section of the settings for your repository.
> - [GitHub docs, about protected branches](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches)
> - [GitHub docs, require status checks before merging](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/defining-the-mergeability-of-pull-requests/about-protected-branches#require-status-checks-before-merging)
>
> If you're using another platform, search their documentation for a similar feature.

If you have any tips/tricks/suggestions, please let me know. 😉
